Better editor experience

DeanPDX opened the pull request:
This is to try to get this to feel more like a web IDE. Closes #28. This adds the following:

- Change editor to Ace editor.
- Show indication we are waiting for the server when user performs an action.
- Detect OS' preference for light/dark theme and default to that.
- Allow users to switch between light/dark.
- On format, like Go playground, actually format code.

It borrows heavily from the Go playground. Here's what it looks like in light mode:

<img width="892" alt="Screenshot 2025-01-26 at 1 16 15 PM" src="https://github.com/user-attachments/assets/db244c3d-8fef-4212-9669-a9b20e3e6d5b" />

Here's dark mode (I tried to find a zig-ish theme; we can also just expose Ace's themes to the users):

<img width="891" alt="Screenshot 2025-01-26 at 1 15 51 PM" src="https://github.com/user-attachments/assets/3edbd861-451a-45c2-8bb9-ddcfe42ed2ff" />

I used a flex-based CSS layout which generally will resize down to phone screens:

<img width="353" alt="Screenshot 2025-01-26 at 1 18 25 PM" src="https://github.com/user-attachments/assets/f961284e-2e06-470e-846d-74022005cca3" />

Anyway, give it a try and tweak it to your liking.

DeanPDX commented:
BTW after doing this I found [this playground](https://zig.fly.dev). Source code is [here](https://github.com/dgv/play.zig) (it mentions this project in the readme). Anyway, it already implements a very Go Playround-ish share feature as well. And I noticed it's *fast*. Give it a try. Might be interesting to learn from what they are doing for performance and/or join forces with that maintainer at some point.

It's using [codemirror](https://codemirror.net) for the editor for what it's worth.

gsquire commented:
That implementation is definitely faster. I disabled the build cache because the instance would run out of disk space quite quickly. Last time I checked, there was no option to limit the build cache size. I would need to revisit this to speed things up...

Anyway, I will review this now! The screenshots look excellent.

gsquire reviewed (commented):
This is a great addition, thank you. Would you mind formatting the HTML? Happy to merge afterwards.

DeanPDX commented:
I updated the HTML. I believe the part you thought looked ugly was due to me trying to preserve the Zig code's indentation. Since I'm setting up the Ace editor anyway I just switched it to being in the playground.js as a constant. This also gets us closer to the Go Playground's select list of various code examples.

gsquire commented:
No worries on the formatting, just being pedantic. I will go ahead and deploy this shortly. They look great locally!

I really appreciate the changes.
